### ISSUE TITLE
Questions regarding the cls loss

### ISSUE THREAD

**Issue by leonard-git-acc** [umautobots/SAD-GS#6]
Hello,
I have been looking through your code for the cls loss and I have come across two places, where I have some questions.
The first place is [here](https://github.com/umautobots/SAD-GS/blob/f9055f741b26d52817abc9bfb454869ce9f15563/train.py#L88). I am not quite sure why the sum of the probabilities is clamped. Is this better, than, for example, normalizing it?
``` python
probs_ = torch.clamp(probs_, 0., 1.)
```

My second question concerns the same file [in this location](https://github.com/umautobots/SAD-GS/blob/f9055f741b26d52817abc9bfb454869ce9f15563/train.py#L261). Here there is the variable `data_balancing_weight` declared as 0 and never assigned a new value. It is then multiplied with `occ_pts_prob_error.mean()`, which then, if I am not missing anything, should not have any impact in the loss calculation. Is this intended?

``` python
data_balancing_weight=0
occ_pts_prob_error = torch.abs(1 - occ_pts_prob)
free_pts_prob_error = torch.abs(0 - free_pts_prob)
            
# L1 LOSS
cls_loss = occ_pts_prob_error.mean()*data_balancing_weight + free_pts_prob_error.mean()
loss += cls_loss
```

Thank you for your help!

**Comment by kungfrank** [comment]
Hello,

Thanks for your questions!

1. I clamp the queried probability rather than normalize it because I wanted to preserve the **occupied probability** at the (x, y, z) position, which carries physical significance. Normalizing it would distort this meaning.

2. You're right. The `data_balancing_weight` was used in a previous experiment, but I found it unhelpful for my current method, so I have disabled it.
